scripts/template_upload_ec2.py

```diff
@@ -96,143 +110,158 @@ def download_image_file(image_url, destination=None):
-    os.system('cls')
-    file_size_dl = 0
-    block_sz = 8192
-    while True:
-        buffer_f = u.read(block_sz)
-        if not buffer_f:
-            break
-
-        file_size_dl += len(buffer_f)
-        f.write(buffer_f)
-    f.close()
+    with open(file_name, 'wb') as image_file:
+        # os.system('cls')
+        file_size_dl = 0
+        block_sz = 8192
+        while True:
+            buffer_f = u.read(block_sz)
+            if not buffer_f:
+                break
+
+            file_size_dl += len(buffer_f)
+            image_file.write(buffer_f)
     return file_name, file_path
 
 
-def create_image(template_name, image_description, bucket_name, key_name):
-    """Imports the Image file uploaded to bucket inside amazon s3, checks the image status.
-       Waits for import_image task to complete. creates name tag and assigns template_name to
-       imported image.
+def create_image(ec2, ami_name, bucket_name):
+    """
+    Create the image from a given bucket+file defined by the ami_name
+    :param ec2: mgmtsystem:EC2System object
+    :param ami_name: name of the file in the bucket, will be used for AMI name too
+    :param bucket_name: name of the s3 bucket where the image is
+    :return: none
+    """
+    print('EC2:{}: Adding image {} from bucket {}...'.format(ec2.api.region, ami_name, bucket_name))
+    import_task_id = ec2.import_image(s3bucket=bucket_name,
+                                      s3key=ami_name,
+                                      description=ami_name)
+    print('EC2:{}: Monitoring image task id {}...'.format(ec2.api.region, import_task_id))
+
+    wait_for(ec2.get_image_id_if_import_completed,
+             func_args=[import_task_id],
+             fail_condition=False,
+             delay=5,
+             timeout='30m')
+
+    ami_id = check_for_ami(ec2, ami_name)
+
+    print("EC2:{}: Creating Tag for image {}...".format(ec2.api.region, ami_name))
 
-        Args:
-            template_name: Name of the template, this will be assigned to Name tag of the
-            imported image.
-            image_description: description to be set to imported image.
-            bucket_name: bucket_name from where image file is imported. This is created in Amazon S3
-            service.
-            key_name: Keyname inside the create bucket.
+    ec2.ec2_connection.create_tags(resource_ids=[ami_id],
+                                   tags={'Name': ami_name})
+
+
+def upload_to_s3(ec2, upload_bucket_name, ami_name, file_path):
+    """
+    Create a bucket and upload the given file to the bucket
+    :param ec2: mgmtsystem:EC2System object
+    :param upload_bucket_name: string name of bucket to create/upload to
+    :param ami_name: string name of the file in the bucket
+    :param file_path: string path to the local file to upload
+    :return: none
+    """
+    print("EC2:{}: Creating bucket {}...".format(ec2.api.region, upload_bucket_name))
+    ec2.create_s3_bucket(upload_bucket_name)
```

Comment:
You are creating for every upload a bucket? So there should be removal of this bucket after successful import into AMI? I can create method for deleting buckets in wrapanapi.